src/index.ts

```diff
@@ -76,4 +79,62 @@ app.post("/logout", async(c) => {
   return new Response();
 });
 
+app.post("/attendance", async(c) => {
+  /*ネットワークの確認を行う*/
+  const allowPattern = new RegExp(c.env.ALLOW_IP_REGEX);
+  if ( !allowPattern.test(c.req.header("cf-connectiong-ip") ?? "") ) { //"??":c.req.header(cf-connectiong-ip")でnullが返ってきたときに""として扱う演算子
+    return c.text("Forbidden", 403);
+  }
+
+  /*学生の認証を行う*/
+  //sessionがcookieあるか確認
+  const honoSession = c.get('session');
+  const session = honoSession.get("login") as Session | null;
+  if ( !session ){
+    return c.text("Unauthorized", 401);
+  }
+
+  /*まとめてクエリを実行する*/
+  const now = new Date();
+  const [ existResult, accountEntry, attendanceBoardEntry ] = await c.env.DB.batch([
+    c.env.DB.prepare("SELECT * FROM session WHERE id = ?").bind(session.id),
+    c.env.DB.prepare("SELECT * FROM account WHERE role = 'STUDENT' AND id = ?").bind(session.account.id),
+    c.env.DB.prepare("SELECT id FROM attendance_board WHERE start_from  <= ?1 AND seconds_from_be_late_to_end >= ?2").bind(now, now),
+  ]);
+  //session情報がsessionテーブルにあるか確認
+  const existsSession = existResult.results.length === 1;
+  if ( !existsSession ){
+    return c.text("Unauthorized", 401);
+  }
+  //roleが学生か確認
+  const isStudent = accountEntry.results.length === 1;
+  if (!isStudent) {
+    return c.text("Unauthorized", 401);
+  }
+
+  /*出席申請受付が開始されている科目をDBから探す、存在しなければ404を返す*/
+  const isOpenBoard = attendanceBoardEntry.result.length === 1;
+  if ( !isOpenBoard ){
+    return c.text("Not Found", 404);
+  }
+
+  /*まとめてクエリを実行する*/
+  const [ attendResult, request ] = await c.env.DB.batch([
+    c.env.DB.prepare("SELECT * FROM attendance WHERE \"where\" = ?").bind(attendanceBoardEntry),
+    c.env.DB.prepare("INSERT INTO attendance (id, create_at, who, \"where\") VALUES (?1, ?2, ?3, ?4)").bind(nanoid(), now, session.account.id, attendanceBoardEntry)
+  ])
+  /*打刻を行う*/
+  //既に出席申請されていないか確認する
+  const existAttendance = attendResult != null;
```

Comment:
```suggestion
  const existAttendance = attendResult.results.length !== 0;
```
こちらも `results` を読み取りましょう.